introduction_to_applying_machine_learning/visual_object_detection/README.md

```diff
@@ -0,0 +1,145 @@
+     | mAP | 0.067 | 0.226 | 0.371 | 0.375|
+
+
+### 1.4. What is the Estimated Cost? <a name="cost"></a>
+
+* Running the notebook costs around $130~140 USD. This notebook provides advanced materials, including finetuning two types of pretrained Sagemaker models **till convergence**, with and without hyperparameter optimization (HPO), and result in four models for inference. You could choose to train either one model, or all four models according to your budget and requirements. The cost and runtime for training each model (assuming using p3.2xlarge EC2 instance) are:
+
+   | Model | Cost (USD) | Runtime (Hours) | Billable time (Hours)|
+   |:----------:|:---------------:|:----:|:-----:|
+   |Type 1|    1.5     |       0.5       | 0.5|
+   |Type 1 with HPO (20 jobs)|    30.6    |       1*        | 10|
+   |Type 2|    4.6     |       1.5       | 1.5|
+   |Type 2 with HPO (20 jobs)|     92     |       3*        | 30|
+  (*) HPO tasks in this solution consider 20 jobs in total and 10 jobs in parallel. So 1 actual runtime hour amounts to 10 billable cost hours.
+* Please make sure you have read the cleaning up part [here](#cleaning-up) after training to avoid incurred cost from deployed models.
+
+
+
+### 1.5. What Algorithms & Models are Used? <a name="algorithms"></a>
+
+* The pretrained Sagemaker models include SSD models and FasterRCNN model, using either VGG, ResNet, or MobileNet as backbone, pretrained on either ImageNet, COCO, VOC, or FPN dataset.
+
+### 1.6. How Does the Data Flow Look Like? <a name="dataflow"></a>
+
+![Data flow](https://sagemaker-solutions-prod-us-east-2.s3.us-east-2.amazonaws.com/sagemaker-defect-detection/docs/data_flow.png)
+
+## 2. Solution Details <a name="solution"></a>
+
+### 2.1. Background <a name="background"></a>
+
+According to the [Gartner](https://www.gartner.com/smarterwithgartner/gartner-top-10-strategic-technology-trends-for-2020/),  hyper-automation is the number one trend in 2020 and will continue advancing in future. When it comes to manufacturing, one of the main barriers to hyper-automation is in areas where Human involvements is still struggling to be reduced and intelligent systems have hard times to become on-par with Human visual recognition abilities and become mainstream, despite great advancement of Deep Learning in Computer Vision. This is mainly due to lack of enough annotated data (or when data is sparse) in areas such as _Quality Control_ sections where trained Human eyes still dominates.
+
+
+### 2.2. What is Visual Inspection? <a name="inspection"></a>
+
+The **analysis of products on the production line for the purpose of Quality Control**. Visual inspection can also be used for internal and external assessment of the various equipment in a production facility such as storage tanks, pressure vessels, piping, and other equipment ([source](https://nanonets.com/blog/ai-visual-inspection/)) which expands to many industries from Electronics, Medical, Food and Raw Materials.
```

Comment:
```suggestion
The **analysis of products on the production line for the purpose of Quality Control**. Visual inspection can also be used for ([internal and external assessment of the various equipment in a production facility such as storage tanks, pressure vessels, piping, and other equipment](https://nanonets.com/blog/ai-visual-inspection/)) which expands to many industries from Electronics, Medical, Food and Raw Materials.
```

Reply:
I changed to the following:

The **analysis of products on the production line for the purpose of Quality Control**. According to [Everything you need to know about Visual Inspection with AI
](https://nanonets.com/blog/ai-visual-inspection/), visual inspection can also be used for internal and external assessment of the various equipment in a production facility such as storage tanks, pressure vessels, piping, and other equipment which expands to many industries from Electronics, Medical, Food and Raw Materials.